pkg/core/interop_neo_test.go

```diff
@@ -0,0 +1,435 @@
+	require.NoError(t, err)
+	assetID := v.Estack().Pop().Value()
+	require.Equal(t, assetState.ID.Bytes(), assetID)
+}
+
+func TestAssetGetAssetType(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetAssetType(v)
+	require.NoError(t, err)
+	assetType := v.Estack().Pop().Value()
+	require.Equal(t, big.NewInt(int64(assetState.AssetType)), assetType)
+}
+
+func TestAssetGetAvailable(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetAvailable(v)
+	require.NoError(t, err)
+	available := v.Estack().Pop().Value()
+	require.Equal(t, big.NewInt(int64(assetState.Available)), available)
+}
+
+func TestAssetGetIssuer(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetIssuer(v)
+	require.NoError(t, err)
+	issuer := v.Estack().Pop().Value()
+	require.Equal(t, assetState.Issuer.Bytes(), issuer)
+}
+
+func TestAssetGetOwner(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetOwner(v)
+	require.NoError(t, err)
+	owner := v.Estack().Pop().Value()
+	require.Equal(t, assetState.Owner.Bytes(), owner)
+}
+
+func TestAssetGetPrecision(t *testing.T) {
+	v, assetState, context := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+
+	err := context.assetGetPrecision(v)
+	require.NoError(t, err)
+	precision := v.Estack().Pop().Value()
+	require.Equal(t, big.NewInt(int64(assetState.Precision)), precision)
+}
+
+func TestRuntimeSerializeDeserialize(t *testing.T) {
+	v := vm.New()
+	v.Estack().PushVal(vm.NewBoolItem(true))
+	err := vm.RuntimeSerialize(v)
+	require.NoError(t, err)
+
+	err = vm.RuntimeDeserialize(v)
+	require.NoError(t, err)
+}
+
+func TestRuntimeSerializeNegative(t *testing.T) {
+	v, assetState, _ := createVMAndAssetState(t)
+	v.Estack().PushVal(vm.NewInteropItem(assetState))
+	err := vm.RuntimeSerialize(v)
+	require.Errorf(t, err, "not supported")
+}
+
+// Helper functions to create VM, InteropContext, TX, AccountState, ContractState, AssetState.
+
+func createVMAndPushInterop(t *testing.T) (*vm.VM, *Block, *interopContext) {
+	v := vm.New()
+	block := newDumbBlock()
+	context := newInteropContext(0x10, newTestChain(t), storage.NewMemoryStore(), block, nil)
+	v.Estack().PushVal(vm.NewInteropItem(block))
+	return v, block, context
+}
+
+func createVMAndPushTX(t *testing.T) (*vm.VM, *transaction.Transaction, *interopContext) {
+	v, tx, context := createVMAndTX(t)
+	v.Estack().PushVal(vm.NewInteropItem(tx))
+	return v, tx, context
+}
+
+func createVMAndAssetState(t *testing.T) (*vm.VM, *AssetState, *interopContext) {
+	v := vm.New()
+	assetState := &AssetState{
+		ID:         util.Uint256{},
+		AssetType:  0,
+		Name:       "",
+		Amount:     0,
```

Comment:
Maybe set this values to some non-default values?

Reply:
sure why not =)